### PR TITLE
[3763] Split the SelectionDialogDescription to prepare the Tree support

### LINKED ISSUE #3763: Make it possible to display the content of the Selection Dialog using a tree viewer
* [x] **I have checked that this feature has not yet been suggested by someone else.**

### Expected behavior

We want to let the specifier to choose if he want to display the semantic candidates as a flat list (current behavior) or as a tree.

This issue is linked to this ADR : https://github.com/eclipse-sirius/sirius-web/pull/3723
...

### Cuttings backs

...

### Potential side effects

...

### Describe alternatives you've considered

...

### PR DESCRIPTION
**Pull request by florianbarbin**
The SelectionDialogDescription will now just hold the message information
The SelectionDialogDescription will now have a
selectionDialogTreeDescription to make it possible for the specifier to display the selectable elements as a tree.
The selectionDialogTreeDescription will be relatively simple at first time.

Bug: #3763

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?

The current SelectionDialog behavior is still the same.

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: acceleoGeneratorSample.patch
Comment (AxelRICHARD) [inline acceleoGeneratorSample.patch:1]: What's this new file?

File: packages/view/backend/sirius-components-view-diagram-edit/src/main/resources/plugin.properties
Comment (AxelRICHARD) [inline packages/view/backend/sirius-components-view-diagram-edit/src/main/resources/plugin.properties:1]: Please let the file as it was or double check that only your new entries have been added and the existing ones have not been modified.

File: acceleoGeneratorSample.patch
Comment (florianbarbin) [reply acceleoGeneratorSample.patch:1]: oups it should not be here

File: packages/view/backend/sirius-components-view-diagram-edit/src/main/resources/plugin.properties
Comment (florianbarbin) [reply packages/view/backend/sirius-components-view-diagram-edit/src/main/resources/plugin.properties:1]: ok

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/migration/participants/SelectionDialogDescriptionSelectionCandidatesExpressionMigrationParticipant.java
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/migration/participants/SelectionDialogDescriptionSelectionCandidatesExpressionMigrationParticipant.java]: Note to myself or whoever ends up merging this PR: this will have to be updated before the final merge

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/services/diagrams/ModelOperationDiagramDescriptionProvider.java
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/services/diagrams/ModelOperationDiagramDescriptionProvider.java]: I'll remove this empty line

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/SelectionDialogTreeDescriptionBuilder.java
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/SelectionDialogTreeDescriptionBuilder.java:21]: This code should be regenerated after a rebase since the generation strategy has been modified. This file will thus end up in another package.

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/DiagramBuilders.java
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/DiagramBuilders.java:364]: Same comment as the one before

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/SelectionDialogDescriptionBuilder.java
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/SelectionDialogDescriptionBuilder.java:50]: Same comment

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/SelectionDialogTreeDescriptionBuilder.java
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/SelectionDialogTreeDescriptionBuilder.java:21]: Same comment

File: packages/view/backend/sirius-components-view-diagram-edit/src/main/java/org/eclipse/sirius/components/view/diagram/provider/DiagramEditPlugin.java
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-diagram-edit/src/main/java/org/eclipse/sirius/components/view/diagram/provider/DiagramEditPlugin.java]: Formatting

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/migration/participants/SelectionDialogDescriptionSelectionCandidatesExpressionMigrationParticipant.java
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/migration/participants/SelectionDialogDescriptionSelectionCandidatesExpressionMigrationParticipant.java:44]: Can't we use a flatMap somewhere?

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
Most of the code looks good, I'll rebase it and fix the tiny issues detected here and there (it should mostly be conflicts with master)

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/DiagramBuilders.java
Comment (sbegaudeau) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/DiagramBuilders.java:364]: I was mistaken, it has already been rebased on top of the commit I was thinking about. There's nothing to change here

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/SelectionDialogTreeDescriptionBuilder.java
Comment (sbegaudeau) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/SelectionDialogTreeDescriptionBuilder.java:21]: I was mistaken, it has already been rebased on top of the commit I was thinking about. There's nothing to change here

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/SelectionDialogDescriptionBuilder.java
Comment (sbegaudeau) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/SelectionDialogDescriptionBuilder.java:50]: I was mistaken, it has already been rebased on top of the commit I was thinking about. There's nothing to change here

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/SelectionDialogTreeDescriptionBuilder.java
Comment (sbegaudeau) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/diagram/SelectionDialogTreeDescriptionBuilder.java:21]: I was mistaken, it has already been rebased on top of the commit I was thinking about. There's nothing to change here

File: packages/view/backend/sirius-components-view-diagram-edit/src/main/java/org/eclipse/sirius/components/view/diagram/provider/DiagramEditPlugin.java
Comment (sbegaudeau) [reply packages/view/backend/sirius-components-view-diagram-edit/src/main/java/org/eclipse/sirius/components/view/diagram/provider/DiagramEditPlugin.java]: Done

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/services/diagrams/ModelOperationDiagramDescriptionProvider.java
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/services/diagrams/ModelOperationDiagramDescriptionProvider.java]: Done

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/migration/participants/SelectionDialogDescriptionSelectionCandidatesExpressionMigrationParticipant.java
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/migration/participants/SelectionDialogDescriptionSelectionCandidatesExpressionMigrationParticipant.java]: Updated

**Comment by sbegaudeau** [comment]
I'll wait for the build and merge it

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**
